openmetadata-ui/src/main/resources/ui/src/components/MyData/RightSidebar/FollowingWidget.tsx

```diff
@@ -10,249 +10,374 @@
+import { FOLLOWING_WIDGET_FILTER_OPTIONS } from '../../../constants/Widgets.constant';
+import { SIZE } from '../../../enums/common.enum';
+import { EntityTabs } from '../../../enums/entity.enum';
+import { SearchIndex } from '../../../enums/search.enum';
 import { EntityReference } from '../../../generated/entity/type';
+import { TagLabel } from '../../../generated/type/tagLabel';
 import { useApplicationStore } from '../../../hooks/useApplicationStore';
 import { SearchSourceAlias } from '../../../interface/search.interface';
 import {
   WidgetCommonProps,
   WidgetConfig,
 } from '../../../pages/CustomizablePage/CustomizablePage.interface';
-import customizeMyDataPageClassBase from '../../../utils/CustomizeMyDataPageClassBase';
+import { searchQuery } from '../../../rest/searchAPI';
 import entityUtilClassBase from '../../../utils/EntityUtilClassBase';
 import { getEntityName } from '../../../utils/EntityUtils';
-import { getUserPath } from '../../../utils/RouterUtils';
+import { getDomainPath, getUserPath } from '../../../utils/RouterUtils';
 import searchClassBase from '../../../utils/SearchClassBase';
 import serviceUtilClassBase from '../../../utils/ServiceUtilClassBase';
-import ErrorPlaceHolder from '../../common/ErrorWithPlaceholder/ErrorPlaceHolder';
-import EntityListSkeleton from '../../common/Skeleton/MyData/EntityListSkeleton/EntityListSkeleton.component';
+import { getUsagePercentile } from '../../../utils/TableUtils';
+import { showErrorToast } from '../../../utils/ToastUtils';
+import EntitySummaryDetails from '../../common/EntitySummaryDetails/EntitySummaryDetails';
+import { OwnerLabel } from '../../common/OwnerLabel/OwnerLabel.component';
+import { SourceType } from '../../SearchedData/SearchedData.interface';
+import TagsV1 from '../../Tag/TagsV1/TagsV1.component';
+import WidgetEmptyState from '../Widgets/Common/WidgetEmptyState/WidgetEmptyState';
+import WidgetFooter from '../Widgets/Common/WidgetFooter/WidgetFooter';
+import WidgetHeader from '../Widgets/Common/WidgetHeader/WidgetHeader';
+import WidgetWrapper from '../Widgets/Common/WidgetWrapper/WidgetWrapper';
 import './following-widget.less';
 
-export interface FollowingWidgetProps extends WidgetCommonProps {
-  followedData: EntityReference[];
-  isLoadingOwnedData: boolean;
-}
-
 function FollowingWidget({
   isEditView,
-  followedData,
-  isLoadingOwnedData,
   handleRemoveWidget,
   widgetKey,
   handleLayoutUpdate,
   currentLayout,
-}: Readonly<FollowingWidgetProps>) {
+}: Readonly<WidgetCommonProps>) {
   const { t } = useTranslation();
   const { currentUser } = useApplicationStore();
   const navigate = useNavigate();
+  const [selectedEntityFilter, setSelectedEntityFilter] = useState<string>(
+    FOLLOWING_WIDGET_FILTER_OPTIONS[0].key
+  );
+  const [followedData, setFollowedData] = useState<SourceType[]>([]);
+  const [isLoadingOwnedData, setIsLoadingOwnedData] = useState<boolean>(false);
+
+  const getSortField = (filterKey: string): string => {
```

Comment:
Let's move this to utils or common widget utils and use enums instead of hard coded strings. Also, let's use keys instead of values.